fix crash with playing playback on RPi

pgorszkowski-igalia opened the pull request:
Without this change WPE crashes when we try to start to play a playback on RPi. The reason is with OMX quirk(GStreamerQuirkOpenMAX) which used playbin flags ("soft-colorbalance") from base class: GStreamerQuirk.
<!--EWS-Status-Bubble-Start-->
https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/4aad871d9dbb575748c10fe46ed8902d587bcfbd

| Build-Tests | Layout-Tests |
| ----------- | ------------ |
| [✅ 🛠 wpe-238-amd64-build](https://ews-wpe-rdk.igalia.com/#/builders/1/builds/39 "Built successfully") | [✅ 🧪 wpe-238-amd64-layout](https://ews-wpe-rdk.igalia.com/#/builders/5/builds/26 "Passed tests") 
| [✅ 🛠 wpe-238-arm32-build](https://ews-wpe-rdk.igalia.com/#/builders/8/builds/38 "Built successfully") | [✅ 🧪 wpe-238-arm32-layout](https://ews-wpe-rdk.igalia.com/#/builders/6/builds/37 "Passed tests") 
| | 
| | 
<!--EWS-Status-Bubble-End-->

pgorszkowski-igalia commented:
I will upstream it first

pgorszkowski-igalia commented:
> The patch looks good to me. At first, I pondered if it would make more sense (conceptually speaking) to set a default of 0 in [GStreamerQuirk::getAdditionalPlaybinFlags()](https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/pgorszkowski/2.38/fix_crash_on_rpi_with_playing_playback/Source/WebCore/platform/gstreamer/GStreamerQuirks.h#L80) and modify all the subclasses to return text and soft-colorbalance, but after thinking a bit more, I don't think the code duplication makes sense. The solution you proposed (just override the method and return 0 for the case of OpenMAX) is simpler (less changes, lower binary size) and equally effective, so I support it.

Actually in upstream [GStreamerQuirk::getAdditionalPlaybinFlags()](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/gstreamer/GStreamerQuirks.h/#L81) returns 0, so the upstream change will look weird.

eocanha commented:
Then my suggestion is that you add the same non-zero default implementation upstream too as part of the upstreaming of your patch.

Nobody actually uses the Quirks system upstream, and the code is there so that downstream can inherit it on the next WPE version upgrade. If downstream default implementation returns non-zero, upstream should also return the same. Otherwise we're going to have trouble downstream on next upgrade.

If you need to justify it in an upstream code review, just point to the non-zero value downstream and to this comment from me.

pgorszkowski-igalia committed:
[GStreamer][Quirks][RPi] update additional playbin flags
https://bugs.webkit.org/show_bug.cgi?id=286668

Reviewed by Philippe Normand.

Set proper flags used by OpenMax.

* Source/WebCore/platform/gstreamer/GStreamerQuirkOpenMAX.h:

Canonical link: https://commits.webkit.org/289549@main
